buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -414,7 +414,7 @@ class BuildPlugin implements Plugin<Project> {
                 File gradleJavaHome = Jvm.current().javaHome
                 // we fork because compiling lots of different classes in a shared jvm can eventually trigger GC overhead limitations
                 options.fork = true
-                options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
+                options.forkOptions.javaHome = new File(project.javaHome)
```

Comment:
How about:

```diff
diff --git a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
index 4f59e8e185..7dbcd66688 100644
--- a/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
+++ b/buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy
@@ -414,7 +414,11 @@ class BuildPlugin implements Plugin<Project> {
                 File gradleJavaHome = Jvm.current().javaHome
                 // we fork because compiling lots of different classes in a shared jvm can eventually trigger GC overhead limitations
                 options.fork = true
-                options.forkOptions.javaHome = new File(project.javaHome)
+                if (GradleVersion.current() >= GradleVersion.version("3.5")) {
+                    options.forkOptions.javaHome = new File(project.javaHome)
+                } else {
+                    options.forkOptions.executable = new File(project.javaHome, 'bin/javac')
+                }
                 options.forkOptions.memoryMaximumSize = "1g"
                 if (project.targetCompatibility >= JavaVersion.VERSION_1_8) {
                     // compile with compact 3 profile by default
```

Reply:
This is one option indeed. Why not use the Gradle wrapper to pin the Gradle to a single version? I checked and 4.4 seems to work fine.

We have some nice [new documentation on using the wrapper](https://docs.gradle.org/nightly/userguide/gradle_wrapper.html), too.

Reply:
We are not going to use the Gradle wrapper, sorry. Relates #13744.

Reply:
Are you interested in taking this PR to completion?

Reply:
Sorry I forgot about this. I'm just perplexed about the decision not to use the Gradle wrapper. To me it simply doesn't make sense. Regardless, given that you want to keep running the build with 3.3 there's no advantage to the added complexity, and therefore we should just drop this PR.

Reply:
Sorry, this is not the place to rehash the decision, and we are unlikely to be swayed anyway.